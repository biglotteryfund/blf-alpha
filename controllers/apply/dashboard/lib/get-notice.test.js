/* eslint-env jest */
'use strict';
const getNotice = require('./get-notice');

test('get notice if pending under Â£10,000 application in England', function () {
    const mockUnder10kEngland = {
        formId: 'awards-for-all',
        applicationData: { projectCountry: 'england' },
    };

    const mockUnder10kEmpty = {
        formId: 'awards-for-all',
        applicationData: null,
    };

    const mockOver10k = {
        formId: 'standard-enquiry',
        applicationData: { projectCountries: ['england'] },
    };

    const resultEn = getNotice('en', [
        mockUnder10kEngland,
        mockUnder10kEmpty,
        mockOver10k,
    ]);

    expect(resultEn).toMatchSnapshot();

    const resultCy = getNotice('cy', [
        mockUnder10kEngland,
        mockUnder10kEmpty,
        mockOver10k,
    ]);

    expect(resultCy).toMatchSnapshot();

    const noResult = getNotice('en', [mockUnder10kEmpty, mockOver10k]);
    expect(noResult).toBeUndefined();
});
