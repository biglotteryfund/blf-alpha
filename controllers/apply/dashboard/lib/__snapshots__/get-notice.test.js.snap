// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get notice if pending under £10,000 application in England 1`] = `
Object {
  "body": "If you've started an application already, and it's not related to supporting your community or organisation through the pandemic, we won't be able to accept it. But you could decide to start a new one that focuses on COVID-19 instead.",
  "title": "For funding under £10,000 in England, we're now only accepting COVID-19 related applications",
}
`;

exports[`get notice if pending under £10,000 application in England 2`] = `
Object {
  "body": "Os ydych chi wedi cychwyn cais yn barod, ac nad yw'n gysylltiedig â chefnogi'ch cymuned neu sefydliad trwy'r pandemig, ni fyddwn yn gallu ei dderbyn. Ond fe allech chi benderfynu cychwyn un newydd sy'n canolbwyntio ar COVID-19 yn lle.",
  "title": "Ar gyfer ariannu o dan £10,000 yn Lloegr, dim ond ceisiadau cysylltiedig â COVID-19 yr ydym yn eu derbyn",
}
`;
